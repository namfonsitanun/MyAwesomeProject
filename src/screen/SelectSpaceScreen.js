import React, { Component } from 'react'
import styled from 'styled-components/native'
import { Image, StyleSheet, Text, View } from 'react-native'

const StyledView = styled.View`
	flex: 1;
	align-items: center;
	padding-top: 20px;
	padding-bottom: 20px;
`

const ContentView = styled.View`
	justify-content: flex-start;
	flex-direction: row;
	flex-wrap: wrap;
	margin: 20px;
`

const StyledText = styled.Text`
	color: ${props => (props.color ? props.color : 'black')};
	text-align: ${props => (props.position || 'center')};
	font-weight: ${props => (props.text || 'normal')};
`
const Card = styled.View`
	border-radius: 5px;
	background-color: white;
	margin: 5px;
`


const Status = styled.View`
	justify-content: flex-end;
	align-items: flex-end;
	border-radius: 50;
	width: 12px;
	height: 12px;
	background-color: ${props => (props.color || 'red')};
`

const Button = styled.TouchableOpacity`
	justify-content: ${props => (props.position || 'center')};
	background-color: #f9253e;
	color: white;
	padding: 10px;
	border-radius: 5px;
	width: ${props => (props.width || '325px')};
`

export class SelectSpaceScreen extends React.Component {
  constructor(props) {
    super(props)
  }

	static navigationOptions = {
	  title: 'Select Space',
	};

	render() {
	  const pic = {
	    url: 'https://www.w3schools.com/howto/img_paris.jpg',
	  }
	  return (
  <StyledView>
    <Button title="+ New Space" width="150px">
      <StyledText color="white" text="bold">
						+ New Space
      </StyledText>
    </Button>
    <ContentView>
      <Card>
        <View style={{ flexDirection: 'row', margin: 5, backgroundColor: 'white' }}>
          <StyledText position="left">Travel</StyledText>
          <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'flex-end' }}>
            <Status color="red" />
          </View>
        </View>
        <Image source={pic} style={{ width: 100, height: 75 }} />
      </Card>
      <Card>
        <View style={{ flexDirection: 'row', margin: 5, backgroundColor: 'white' }}>
          <StyledText position="left">Travel</StyledText>
          <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'flex-end' }}>
            <Status color="orange" />
          </View>
        </View>
        <Image source={pic} style={{ width: 100, height: 75 }} />
      </Card>
    </ContentView>
    <View style={{ flex: 1, justifyContent: 'flex-end' }}>
      <Button>
        <StyledText color="white" text="bold">
						1 steps to go >
        </StyledText>
      </Button>
    </View>
  </StyledView>
	  )
	}
}
