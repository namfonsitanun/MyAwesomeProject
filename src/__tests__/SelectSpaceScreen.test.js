import React from 'react'
import 'react-native'
import SelectSpaceScreen from '../screen/SelectSpaceScreen.js'
import renderer from 'react-test-renderer'
import 'styled-components-test-utils/lib/jest'

describe('<SelectSpaceScreen>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SelectSpaceScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
