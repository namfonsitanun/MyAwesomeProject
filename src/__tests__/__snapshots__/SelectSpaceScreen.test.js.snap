// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectSpaceScreen> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 20,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#f9253e",
        "borderRadius": 5,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "width": 150,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      color="white"
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
      text="bold"
    >
      + New Space
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderRadius": 5,
            "marginBottom": 5,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 5,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flexDirection": "row",
            "margin": 5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          position="left"
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "normal",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          Travel
        </Text>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            color="red"
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "backgroundColor": "red",
                  "borderRadius": 50,
                  "height": 12,
                  "justifyContent": "flex-end",
                  "width": 12,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <Image
        source={
          Object {
            "url": "https://www.w3schools.com/howto/img_paris.jpg",
          }
        }
        style={
          Object {
            "height": 75,
            "width": 100,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderRadius": 5,
            "marginBottom": 5,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 5,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flexDirection": "row",
            "margin": 5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          position="left"
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "normal",
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          Travel
        </Text>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            color="orange"
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "backgroundColor": "orange",
                  "borderRadius": 50,
                  "height": 12,
                  "justifyContent": "flex-end",
                  "width": 12,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <Image
        source={
          Object {
            "url": "https://www.w3schools.com/howto/img_paris.jpg",
          }
        }
        style={
          Object {
            "height": 75,
            "width": 100,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f9253e",
          "borderRadius": 5,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": 325,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        color="white"
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontWeight": "bold",
              "textAlign": "center",
            },
            undefined,
          ]
        }
        text="bold"
      >
        1 steps to go &gt;
      </Text>
    </View>
  </View>
</View>
`;
